readthedocs/doc_builder/templates/doc_builder/conf.py.tmpl

```diff
@@ -58,15 +58,39 @@ using_rtd_theme = (
     ) or 'html_theme' not in globals()
 )
 if using_rtd_theme:
-    theme = importlib.import_module('{{ html_theme_import }}')
     html_theme = '{{ html_theme }}'
     html_style = None
     html_theme_options = {}
+
+{% comment %}
+# Legacy behavior for html_theme_path:
+# It is suspected that old builds with sphinx_rtd_theme required an extra
+# definition of html_theme_path in order to work.
+# This behavior is seen documented for old releases of sphinx_rtd_theme.
+# An example of old instructions:
+# https://github.com/readthedocs/sphinx_rtd_theme/blob/abd951a9cd98851269b8b5f5e59b1b4d29b1416c/README.rst
+# Discussion of issue:
+# https://github.com/readthedocs/readthedocs.org/pull/9654
+#
+# We need to isolate the old legacy behavior because it causes Sphinx
+# to skip calling the theme's setup(app) so its initialization never happens.
+{% endcomment %}
+# This following legacy behavior will gradually be sliced out until its deprecated and removed.
+# Skipped for Sphinx 6+
+# Skipped by internal Feature flag SKIP_SPHINX_HTML_THEME_PATH
+# Skipped by all new projects since SKIP_SPHINX_HTML_THEME_PATH's introduction (jan 2023)
```

Comment:
@agjohnson I believe we should also do this, just want to make sure that you have seen it. New projects won't follow strange old install guides, and they are likely on later versions of Sphinx.